File: python/command/commands/light_off_command.py
from external.light import Light
from commands.command import Command


class LightOffCommand(Command):

    def __init__(self, light: Light) -> None:
        self.light = light

    def execute(self) -> None:
        self.light.off()

    def undo(self) -> None:
        self.light.on()

File: java/patterns_example/src/main/java/com/app/patterns/command/commands/LightOffCommand.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.app.patterns.command.commands;

import com.app.patterns.command.external.Light;


public class LightOffCommand implements Command {
    Light light;
    
    public LightOffCommand(Light light) {
        this.light = light;
    }
    
    @Override
    public void execute() {
        light.off();
    }
    
    @Override
    public void undo(){
        light.on();
    }
}